Modules/CaPTkCommon/src/CaPTkROIConstructionHelperBase.h

```diff
@@ -0,0 +1,78 @@
+#ifndef CaPTkROIConstructionHelperBase_h
+#define CaPTkROIConstructionHelperBase_h
+
+// The following header file is generated by CMake and thus it's located in
+// the build directory. It provides an export macro for classes and functions
+// that you want to be part of the public interface of your module.
+#include <MitkCaPTkCommonExports.h>
+
+#include <mitkLabelSetImage.h>
+
+#include <string>
+#include <vector>
+
+namespace captk
+{
+/** \class ROIConstructionHelperBase
+ * \brief Non-templated class to serve as base class for ROIConstructionHelper
+ */
+class MITKCAPTKCOMMON_EXPORT ROIConstructionHelperBase
+{
+public:
+    ROIConstructionHelperBase() {}
+
+    virtual ~ROIConstructionHelperBase() {}
+
+    virtual void Update(
+        float /*radius*/,
+        float /*step*/)
+    {
+        // Cannot have pure virtual because we want to use the base class
+    }
+
+    virtual void GoToBegin()
+    {
+
+    }
+
+    virtual bool IsAtEnd()
+    {
+        return true; // Return is required and we don't want this to be pure virtual
+    }
+
+    virtual void SetValuesAndNames(
+        std::vector<int>         /*values*/,
+        std::vector<std::string> /*names*/)
+    {
+        
+    }
+
+    /** \brief ++ overloading to go to the next lattice */
+    ROIConstructionHelperBase &operator++() // prefix
+    {
+        this->OnIncrement(); // call actual operation function
+        return *this;
+    }
+
+    /** \brief ++ overloading to go to the next lattice */
+    ROIConstructionHelperBase operator++(int) // postfix (calls suffix)
```

Comment:
int isn't used at all in the function?? Do you want to change the signature?

Reply:
operator++() is ```++variable```
operator++(int) is ```variable++```

int is a dummy variable that does nothing, except discriminating between the two. This is a thing from C and we can't do anything about it

Reply:
okay..I remember